### PR TITLE
[DOCS] Rewrite dis max query

### PR DESCRIPTION
**Pull request by jrodewig**
Rewrites the `dis_max` query to use the new query format.

This is part of #40977, an effort to standardize documentation for query types.

## Before
<details>
 <summary>Before image</summary>
Before Image
</details>

## After
<details>
 <summary>After image</summary>
After Image
</details>

### CODE REVIEW

File: docs/reference/query-dsl/dis-max-query.asciidoc [hunk 1]
```diff
@@ -1,48 +1,54 @@
...
-default `tie_breaker` is `0.0`.
-
-This query maps to Lucene `DisjunctionMaxQuery`.
+=== Disjunction Max Query
+
+Returns documents matching one or more wrapped queries, called query clauses or
+clauses.
+
+If a returned document matches multiple query clauses, the `dis_max` query
+assigns the document the highest relevance score from any matching clause.
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/dis-max-query.asciidoc]: I think it is important to add here "plus a tie breaking increment for any additional matching subqueries".

File: docs/reference/query-dsl/dis-max-query.asciidoc [hunk 2]
```diff
@@ -1,48 +1,54 @@
...
++
+--
+(float) Floating point number between `0` and `1.0` used to increase the
+<<query-filter-context, relevance scores>> of documents matching multiple query
+clauses. Defaults to `0.0`.
+
+If a document matches multiple clauses, the `dis_max` query calculates the
+relevance score for the document as follows:
+
+. Take the highest relevance score from a matching clause.
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/dis-max-query.asciidoc]: I think this is a little bit more clear: "take the score from a matching clause with the highest relevance score"

File: docs/reference/query-dsl/dis-max-query.asciidoc [hunk 3]
```diff
@@ -1,48 +1,54 @@
...
+. Multiply the score from any other matching clauses by the `tie_breaker` value.
+. Add the highest score to the multiplied scores and normalize.
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/dis-max-query.asciidoc]: what is `normalize` here?

**Review by mayya-sharipova — CHANGES_REQUESTED** [review]
@jrodewig Thanks! This is indeed a nice restructuring.  But we lost the use case for this query. Do we plan to put the use case somewhere else in the documentation?

**Comment by jrodewig** [comment]
Thanks for the review @mayya-sharipova! I may have gotten carried away with edits so I re-added some context for the use case and example with 6fa3e0c7602.

File: docs/reference/query-dsl/dis-max-query.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/query-dsl/dis-max-query.asciidoc]: Re-added with 6fa3e0c.

File: docs/reference/query-dsl/dis-max-query.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/query-dsl/dis-max-query.asciidoc]: Added with 6fa3e0c.

File: docs/reference/query-dsl/dis-max-query.asciidoc (hunk 3 above)
Comment (jrodewig) [reply docs/reference/query-dsl/dis-max-query.asciidoc]: Removed this with 6fa3e0c.

File: docs/reference/query-dsl/dis-max-query.asciidoc [hunk 4]
```diff
@@ -1,48 +1,65 @@
...
+
+If a returned document matches multiple query clauses, the `dis_max` query
+assigns the document the highest relevance score from any matching clause, plus
+a tie breaking increment for any additional matching subqueries.
+
+This is useful when searching for a word in multiple fields with different boost
+factors. If the query is "albino elephant," this ensures that "albino" matching
+one field and "elephant" matching another gets a higher score than "albino"
+matching both fields.
+
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/dis-max-query.asciidoc]: @jrodewig Thanks for quickly addressing the feedback.  
One small comment, the way information is presented now looks like the query in the example will address what you described before it:  " If the query is "albino elephant" this ensures that "albino" matching one field and "elephant" matching another gets a higher score than "albino" matching both fields.". But for this, we need a slightly different query: bool with two should dismax clauses for each field.  

May be better, would be to remove this sentence about "albino elephant" all together? WDYT?

Comment (jrodewig) [reply docs/reference/query-dsl/dis-max-query.asciidoc]: @mayya-sharipova  Thanks again for your kind review!

I think you're right about removing the sentence. I also update the example request to fit the use case without introducing a more complex bool query.

**Review by mayya-sharipova — APPROVED** [review]
@jrodewig Thanks for your patience on this.

**Comment by jrodewig** [comment]
@elasticmachine run elasticsearch-ci/oss-distro-docs

**Pull request merged**